client/src/components/ProductsContainer/index.jsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+import Container from "@mui/material/Container";
+import SectionTitle from "../common/SectionTitle";
+import ProductCard from "./ProductCard";
+import Pagination from "@mui/material/Pagination";
+
+import "./style.css";
+
+function ProductsContainer({ products }) {
+  const [page, setPage] = useState(1);
+  return (
+    <Container maxWidth="md">
+      <SectionTitle content="All Product" />
+      <section className="products-container">
+        {products.map((product, index) => {
+          if (index + 1 > page * 6 - 6 && index + 1 <= page * 6) {
+            return (
+              <ProductCard
+                key={index}
+                productImage={product.img}
+                productName={product.name}
+                productPrice={product.price}
+              />
+            );
+          }
+        })}
+      </section>
+      <Pagination
```

Comment:
maybe you need to render it conditionally when products more than six,
also, try to list 15 products, see what will happen?!